### PR TITLE
Change references to "frs" -> "azure" in service-specific client packages

### PR DESCRIPTION
**Pull request by sumedhb1995**
The `@fluid-experimental/frs-client` package for connecting with the Azure Fluid Relay service has been renamed to now be `@fluid-experimental/azure-client`. This also comes with the following name changes for the exported classes and interfaces from the package:
- `FrsClient` -> `AzureClient`
- `FrsAudience` -> `AzureAudience`
- `IFrsAudience` -> `IAzureAudience`
- `FrsMember` -> `AzureMember`
- `FrsConnectionConfig` -> `AzureConnectionConfig`
- `FrsContainerConfig` -> `AzureContainerConfig`
- `FrsResources` -> `AzureResources`
- `FrsAzFunctionTokenProvider` -> `AzureFunctionTokenProvider`
- `FrsUrlResolver` -> `AzureUrlResolver`

This PR also includes:
- updates to the documentation as there were a number of class and interface names that also got updated along with the package name itself.
- change in naming and package location from "@fluid-experimental" to "@fluidframework" to move the package out of experimental state

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+## 0.46 Breaking changes
+- [FrsClient has been renamed to AzureClient](#FrsClient-has-been-renamed-to-AzureClient)
```
Comment (heliocliu) [inline BREAKING.md:2]: nit: the # link needs to be all lowercase to work i think

Comment (sumedhb1995) [reply BREAKING.md:2]: The `Property-removed-from-IFluidDataStoreRuntime` in the 0.45 changes has some uppercase but looks like its still working

File: docs/content/docs/deployment/azure-frs.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ const schema = {
     },
     dynamicObjectTypes: [ /*...*/ ],
 }
-const frsClient = new FrsClient(config);
+const frsClient = new AzureClient(config);
```
Comment (heliocliu) [inline docs/content/docs/deployment/azure-frs.md:53]: should these variable names be changed too?

File: docs/content/docs/start/quick-start.md [hunk 3]
```diff
@@ -69,19 +69,19 @@ local in-memory service called Tinylicious, which runs on port 7070 by default.
...
 orderer, and storage URLs that were provided as part of the onboarding process. Then pass that configuration object into
-the `FrsClient` constructor:
+the `AzureClient` constructor:
 
 ```typescript
-// This configures the FrsClient to use a remote Azure Fluid Service instance.
-const config: FrsConnectionConfig = {
+// This configures the AzureClient to use a remote Azure Fluid Service instance.
+const config: AzureConnectionConfig = {
     tenantId: "myFrsTenantId",
```
Comment (heliocliu) [inline docs/content/docs/start/quick-start.md:77]: same comment on naming here.  should these also be changed to "Azure"?  is "FRS" still a valid abbreviation in any circumstances?

File: docs/content/docs/deployment/azure-frs.md (hunk 2 above)
Comment (sumedhb1995) [reply docs/content/docs/deployment/azure-frs.md]: Good catch! Updated, also updating the references in testing.md rn

File: experimental/framework/azure-client/README.md [hunk 4]
```diff
@@ -1,24 +1,24 @@
-# @fluid-experimental/frs-client
+# @fluid-experimental/azure-client
 
-The frs-client package provides a simple and powerful way to consume collaborative Fluid data with the Frs service.
+The azure-client package provides a simple and powerful way to consume collaborative Fluid data with the Frs service.
```
Comment (heliocliu) [inline experimental/framework/azure-client/README.md:3]:
```suggestion
The azure-client package provides a simple and powerful way to consume collaborative Fluid data with the Azure Fluid Relay service.
```

File: experimental/framework/azure-client/README.md [hunk 5]
```diff
@@ -90,7 +90,7 @@ const schema = {
     },
     dynamicObjectTypes: [ /*...*/ ],
 }
-const frsClient = new FrsClient(config);
+const frsClient = new AzureClient(config);
```
Comment (sdeshpande3) [inline experimental/framework/azure-client/README.md]: Do we want to change the variable name (frsClient) here

File: experimental/framework/azure-client/src/AzureClient.ts [hunk 6]
```diff
@@ -52,42 +52,42 @@ export class FrsClient {
...
     private async getFluidContainerAndServices(
         container: Container,
-    ): Promise<FrsResources> {
+    ): Promise<AzureResources> {
         const rootDataObject = await requestFluidObject<RootDataObject>(container, "/");
         const fluidContainer: FluidContainer = new FluidContainer(container, rootDataObject);
-        const containerServices: FrsContainerServices = this.getContainerServices(container);
-        const frsResources: FrsResources = { fluidContainer, containerServices };
+        const containerServices: AzureContainerServices = this.getContainerServices(container);
+        const frsResources: AzureResources = { fluidContainer, containerServices };
```
Comment (sdeshpande3) [inline experimental/framework/azure-client/src/AzureClient.ts:69]: Here as well -> frsResources

File: experimental/framework/azure-client/src/index.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ export { InsecureTokenProvider } from "@fluidframework/test-runtime-utils";
 export { ITokenProvider } from "@fluidframework/routerlicious-driver";
 
 export * from "./interfaces";
```
Comment (heliocliu) [inline experimental/framework/azure-client/src/index.ts:9]: nit: alphabetize

File: experimental/framework/azure-client/src/interfaces.ts [hunk 8]
```diff
@@ -11,45 +11,45 @@ import {
...
     orderer: string;
     storage: string;
     tokenProvider: ITokenProvider;
 }
 
 /**
- * FrsContainerServices is returned by the FrsClient alongside a FluidContainer.
- * It holds the functionality specifically tied to the Frs service, and how the data stored in
+ * AzureContainerServices is returned by the AzureClient alongside a FluidContainer.
+ * It holds the functionality specifically tied to the Azure Relay Service, and how the data stored in
```
Comment (heliocliu) [inline experimental/framework/azure-client/src/interfaces.ts:28]: I'm assuming it's for brevity here so the whole comment doesn't need to be reformatted, but is the full correct term still "Azure Fluid Relay service" as per the docs?

File: experimental/framework/azure-client/README.md (hunk 5 above)
Comment (sumedhb1995) [reply experimental/framework/azure-client/README.md]: Updated!

File: docs/content/docs/start/quick-start.md (hunk 3 above)
Comment (sumedhb1995) [reply docs/content/docs/start/quick-start.md]: Dropped the "frs" prefix on them 😄 And I believe the term "FRS" itself is still used to refer to the service but it isn't the "official" name any longer whenever we refer to it in code

Comment (sumedhb1995) [reply docs/content/docs/start/quick-start.md]: Also, for the test scripts, is it okay if I update "test:realsvc:frs" to  "test:realsvc:azure" or will that break a build?

**Comment by heliocliu** [comment]
There are still some unaddressed cases you can see from the code surrounding the changes here, but it looks like you'll need to do a search for "frs" throughout the code base rather than known derived terms in order to catch all the usages.

Also for my own education, is the idea behind the naming that they are sufficiently concise within the context of Fluid (even if they might not be in a broader context)?

File: experimental/framework/azure-client/src/interfaces.ts (hunk 8 above)
Comment (sumedhb1995) [reply experimental/framework/azure-client/src/interfaces.ts]: Yep, updated to be the full term + abbreviation

**Comment by sumedhb1995** [comment]
> There are still some unaddressed cases you can see from the code surrounding the changes here, but it looks like you'll need to do a search for "frs" throughout the code base rather than known derived terms in order to catch all the usages.
> 
> Also for my own education, is the idea behind the naming that they are sufficiently concise within the context of Fluid (even if they might not be in a broader context)?

Thanks for catching some of those!! Appreciate it 🙇 Did a scan through with "frs" and all the ones pertaining to the client package should now be changed. The couple remaining ones (like in Gateway or PropertyDDS example) don't need to be updated here.

And right, FRS isn't the "official" brand name for the service when it'll be available on the Azure portal. The idea is that since picking the right client package (in Fluid context) is dependent on what BE service you want to connect to, when you're using `Azure`, you can use the `AzureClient`.

**Review by heliocliu — APPROVED** [review]

**Comment by tylerbutler** [comment]
Why rename this package twice? Can't we switch to the @fluidframework scope rather than leaving it in fluid-experimental? If we can't switch it all the way right now, then we should wait until we can. Each of these renames represents work for customers to update their imports and deps.

**Comment by SamBroner** [comment]
Is Azure to general? E.g. is AzureMember clear enough for folks to understand what that class entails?

**Comment by sumedhb1995** [comment]
> Why rename this package twice? Can't we switch to the @fluidframework scope rather than leaving it in fluid-experimental? If we can't switch it all the way right now, then we should wait until we can. Each of these renames represents work for customers to update their imports and deps.

It would only affect customers if there was a release in between. I was planning on having that name change and directory move for the package to /packages/framework as the direct follow-up to this, in order to keep the PRs modular, but I can include it here as well

**Comment by sumedhb1995** [comment]
> Is Azure to general? E.g. is AzureMember clear enough for folks to understand what that class entails?

Sam! 😄 Good to see you!!
And that's a totally valid concern, its always a balancing act between verbosity and generality. If the import is ultimately framed like `import { AzureMember } from "@fluidframework/azure-client"`, I'm thinking it should be clear that its a Azure member for a Fluid session due to the package naming and scope. There has been some pushback regarding appending `Fluid` to all of our names for verbosity, and this matches up with the proposed naming for the client itself to `AzureClient`

**Comment by sumedhb1995** [comment]
@tylerbutler @heliocliu Also included in the changes to update the package scope to "@fluidframework" and move it out of the experimental folder so that we can include all renaming changes together as requested

**Comment by tylerbutler** [comment]
> @tylerbutler @heliocliu Also included in the changes to update the package scope to "@fluidframework" and move it out of the experimental folder so that we can include all renaming changes together as requested

Thanks! You made a good point regarding releases and the fact that the package wouldn't be published even if you committed and you'd make the rename changes in a later PR.

That said, I appreciate you updated this to include the full rename. I don't think there's a ton of value in having them separate in change history.

Also, sorry for injecting some entropy from vacation land. It's chaos over here! 😋

**Comment by SamBroner** [comment]
> > Is Azure to general? E.g. is AzureMember clear enough for folks to understand what that class entails?
> 
> Sam! 😄 Good to see you!!
> And that's a totally valid concern, its always a balancing act between verbosity and generality. If the import is ultimately framed like `import { AzureMember } from "@fluidframework/azure-client"`, I'm thinking it should be clear that its a Azure member for a Fluid session due to the package naming and scope. There has been some pushback regarding appending `Fluid` to all of our names for verbosity, and this matches up with the proposed naming for the client itself to `AzureClient`

Great point. Thanks Sumedh!

**Comment by sumedhb1995** [comment]
> > @tylerbutler @heliocliu Also included in the changes to update the package scope to "@fluidframework" and move it out of the experimental folder so that we can include all renaming changes together as requested
> 
> Thanks! You made a good point regarding releases and the fact that the package wouldn't be published even if you committed and you'd make the rename changes in a later PR.
> 
> That said, I appreciate you updated this to include the full rename. I don't think there's a ton of value in having them separate in change history.
> 
> Also, sorry for injecting some entropy from vacation land. It's chaos over here! 😋

Not a problem at all! Hope you're having a blast and that the vacation is going swimmingly! 🏖️

**Review by sonalivdeshpande — APPROVED** [review]

**Pull request merged**